ts/src/coindcx.ts

```diff
@@ -0,0 +1,1754 @@
+        //
+        return await this.privatePostExchangeV1OrdersCancelByIds (this.extend (request, params));
+    }
+
+    parseOrder (order, market: Market = undefined): Order {
+        //
+        // privatePostExchangeV1OrdersCreate
+        //     {
+        //         "id": "fcaae278-2a73-11ef-a2d5-5374ccb4f829",
+        //         "client_order_id": null,
+        //         "order_type": "market_order",
+        //         "side": "buy",
+        //         "status": "open",
+        //         "fee_amount": 0.0,
+        //         "fee": 0.0,
+        //         "maker_fee": 0.0,
+        //         "taker_fee": 0.0,
+        //         "total_quantity": 0.0001,
+        //         "remaining_quantity": 0.0001,
+        //         "source": "web",
+        //         "base_currency_name": null,
+        //         "target_currency_name": null,
+        //         "base_currency_short_name": null,
+        //         "target_currency_short_name": null,
+        //         "base_currency_precision": null,
+        //         "target_currency_precision": null,
+        //         "avg_price": 0.0,
+        //         "price_per_unit": 65483.14,
+        //         "stop_price": 0.0,
+        //         "market": "BTCUSDT",
+        //         "time_in_force": "good_till_cancel",
+        //         "created_at": 1718386312000,
+        //         "updated_at": 1718386312000,
+        //         "trades": null // todo check
+        //     }
+        //
+        // privatePostExchangeV1MarginCreate
+        //     {
+        //         "id": 104525583,
+        //         "order_type": "market_order",
+        //         "status": "filled",
+        //         "market": "BTCUSDT",
+        //         "side": "buy",
+        //         "avg_price": 67021.58,
+        //         "total_quantity": 0.0001,
+        //         "remaining_quantity": 0.0,
+        //         "price_per_unit": 0.0,
+        //         "timestamp": 1718650490144.9124,
+        //         "maker_fee": 0.1,
+        //         "taker_fee": 0.1,
+        //         "fee": 0.1,
+        //         "fee_amount": 0.00670216,
+        //         "filled_quantity": 0.0001,
+        //         "bo_stage": "stage_entry",
+        //         "cancelled_quantity": 0.0,
+        //         "stop_price": 0.0
+        //     }
+        //
+        const marketId = this.safeString (order, 'market');
+        market = this.safeMarket (marketId, market);
+        let timestamp = this.safeInteger (order, 'created_at');
+        let datetime: Str = undefined;
+        if (timestamp === undefined) {
+            datetime = this.safeString (order, 'created_at');
+            if (datetime !== undefined) {
+                timestamp = this.parse8601 (datetime);
+            } else {
+                const timestampString = this.safeString (order, 'timestamp', '');
+                const parts = timestampString.split ('.');
+                timestamp = this.parseNumber (parts[0]);
```

Comment:
first we should never use parseNUmber/safeNumber with timestamps, always parseToInt/safeInteger.

Second, instead of `parts[0]` I would rather do `this.safeString(parts, 0)` to avoid crashing the code if the timestamp parsing fails